### ISSUE TITLE
Firefox 77.0.1 scrambles pages

### ISSUE THREAD

**Issue by accetto** [accetto/xubuntu-vnc#7]
Unfortunatelly it has been found, that **Firefox 77.0.1** in the containers from the [release 20.06](https://github.com/accetto/xubuntu-vnc/releases/tag/v20.06) scrambles web pages. Therefore I'll remove all the related images and come with at least some way arround as quickly as possible. Sorry for the inconvenience.

![Screenshot_Firefox_77 0 1_scrambles_pages](https://user-images.githubusercontent.com/34798830/84267799-32e94500-ab16-11ea-88f0-50b5960cbc40.png)

I've filed this bug also in [Bugzilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1644815).

**Comment by accetto** [comment]
Mitigated in the release **20.06.1** by rolling **Firefox** back to the version **76.0.1**.

For the case that you get the message `You've launched an older version of Firefox` please check [Firefox version compatibility](https://github.com/accetto/xubuntu-vnc/wiki/Firefox-version-compatibility) in Wiki.

In essence it explains, that if you want to keep using your previous Firefox profile, you have to replace the content of its- **compatibility.ini** file. 

If it looks like this

```
[Compatibility]
LastVersion=77.0.1_20200602222727/20200602222727
LastOSABI=Linux_x86_64-gcc3
LastPlatformDir=/usr/lib/firefox
LastAppDir=/usr/lib/firefox/browser
```

then replace it with this:

```
[Compatibility]
LastVersion=76.0.1_20200507114007/20200507114007
LastOSABI=Linux_x86_64-gcc3
LastPlatformDir=/usr/lib/firefox
LastAppDir=/usr/lib/firefox/browser
```

You can also create and use a new profile, of course. With the **FF Profile Manager**, which you can find on the desktop, you can easily switch between several profiles.

**Comment by accetto** [comment]
I've found, that the problem relates to the switch FORCE_DISABLE_E10S=1 that I use in containers.

The Docker image tagged as **multiprocess** does not use the switch and it does not scramble web pages. However, Firefox crashes all the time.

With this knowledge I've quickly found also the following pages:

[Almost every site is broken in Firefox 77](https://support.mozilla.org/en-US/questions/1289333)

[Content-Encoding not honored when e10s is turned off via browser.tabs.remote.autostart=false or MOZ_FORCE_DISABLE_E10S=1](https://bugzilla.mozilla.org/show_bug.cgi?id=1638652)

**Comment by accetto** [comment]
Another possible way around is building an image including a newer beta-version of Firefox (e.g. [78.0b6](http://releases.mozilla.org/pub/firefox/releases/78.0b6/)). I've tested that it helps. However, I don't want to use beta-versions in the master branch and releases.

**Comment by accetto** [comment]
Because this is already the second time I had to roll-back Firefox to its previous version and in both cases it was because of multi-process, I'm currently working on images that will not force single-process any more.

Mozilla people want to stop supporting the **FORCE_DISABLE_E10S** switch anyhow. I've planned this change for later, but two forced roll-backs are already enough for me.

If Firefox will be released in its next version before I'm finished with my testing, I'll still release my images with that new Firefox version. If the scrambling problem will be gone, of course.

In any case, the future images that include Firefox will use multi-process by default. The current  off-broadway tag `multiprocess` will be replaced by a new `singleprocess` one.

If Mozilla people are right, the change to multi-process should bring not only better performance by browsing Internet, but also higher security because of sand-boxing. Hopefully it'll work reliably also in Docker.

**Comment by accetto** [comment]
I've just published a new page [Firefox multi-process](https://github.com/accetto/xubuntu-vnc/wiki/Firefox-multiprocess) in Wiki, which explains how the problem will be addressed in future releases.

**Comment by accetto** [comment]
Fixed in the release 20.06.2 by switching to multi-process by default. Be aware that larger shared memory is required. At least 256MB is recommended.

Remark: The next Firefox version (78.0) should not scramble pages even in single-process mode.
